qutebrowser/config/configdata.yml

```diff
@@ -1173,6 +1173,44 @@ editor.encoding:
   default: utf-8
   desc: Encoding to use for the editor.
 
+## fileselect
+
+fileselect.handler:
+  default: default
+  type:
+    name: String
+    valid_values:
+      - default: "Use the default file selector."
+      - external: "Use an external command."
+  desc: Handler for selecting file(s) in forms.
+
+fileselect.single_file.command:
+  type:
+    name: ShellCommand
+    placeholder: true
+  default: ['xterm', '-e', 'export COLUMNS= LINES=;ranger --choosefile={}']
```

Comment:
I'm guessing you got the `export COLUMNS= LINES=` from https://github.com/qutebrowser/qutebrowser/pull/4477#issuecomment-451069864 - right? I've just tried this with `ranger` and it displays as 80x25 despite this. Not sure why that's the case, but given that the `export COLUMNS= LINES=;` doesn't actually seem to help, I suppose we might as well just drop it in the default value here?

Reply:
I haven't actually tried it with ranger as of yet, but with `vifm`. Do you have a preference with `ranger` or could we also have the default use `vifm`? For example as:
```
['xterm', '-e', 'vifm', '--choose-file', '{}']
```

Reply:
No strong preferences from my side here. I'm guessing ranger is much more popular:

- GitHub stars: [ranger](https://github.com/ranger/ranger/) 8.9k, [vifm](https://github.com/vifm/vifm) 1.4k
- Archlinux package stats: [ranger](https://pkgstats.archlinux.de/packages/ranger) ~20%, [vifm](https://pkgstats.archlinux.de/packages/vifm) ~8.5%
- Debian package stats: [ranger](https://qa.debian.org/popcon.php?package=ranger) 1.02%, [vifm](https://qa.debian.org/popcon.php?package=vifm) 0.25%

So some 2-6x times more common - at the same time, I suppose the default here isn't really a suitable default for most people anyways, and many will probably end up changing it (to e.g. use another terminal rather than `xterm`).

I'd kinda prefer keeping ranger because it's probably the more suitable starting point for most people, but if you want to go for vifm instead, I'm not going to stop you :slightly_smiling_face: 

Maybe it'd be a good idea to add a `completions:` to be able to complete ranger/vifm in xterm though?

Reply:
Sounds good :+1:! I added completions to the commands. Let me know if that's the way to add completions and the way you meant :)

Reply:
Hmm, it gives an error. I was looking at `hints.chars` for how to add `completions:` but maybe there is something missing?